lab_4_retrieval_w_clustering/main.py

```diff
@@ -26,6 +29,11 @@ def get_paragraphs(text: str) -> list[str]:
     Returns:
         list[str]: Paragraphs from document.
     """
+    if not text:
+        raise ValueError
+    text = text.replace('\n', ' ')
+    paragraphs = re.split(r'\s{2,}', text.strip())
```

Comment:
oh, so many strings for so many operations. You can just split input text and then iterate over the list and remove empty strings

Reply:
ok, fixed!!!